src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -269,24 +264,20 @@ public void leaveToken(DetailAST ast) {
                 break;
             case TokenTypes.EXPR:
                 // Switch labeled expression has no slist
-                if (parentAst.getType() == TokenTypes.SWITCH_RULE) {
-                    prevScopeUninitializedVariableData = prevScopeUninitializedVariables.peek();
-                    if (shouldUpdateUninitializedVariables(parentAst)) {
-                        updateAllUninitializedVariables(prevScopeUninitializedVariableData);
-                    }
+                if (parentAst.getType() == TokenTypes.SWITCH_RULE
+                    && shouldUpdateUninitializedVariables(parentAst)) {
+                    updateAllUninitializedVariables();
                 }
                 break;
             case TokenTypes.SLIST:
-                prevScopeUninitializedVariableData = prevScopeUninitializedVariables.peek();
                 boolean containsBreak = false;
                 if (parentAst.getType() != TokenTypes.CASE_GROUP
                     || findLastCaseGroupWhichContainsSlist(parentAst.getParent()) == parentAst) {
                     containsBreak = scopeStack.peek().containsBreak;
                     scope = scopeStack.pop().scope;
-                    prevScopeUninitializedVariables.pop();
```

Comment:
It shouldn't cause a problem because of the equivalent push in the `visitToken`. The popped value always contains empty previous data list at this point.

Reply:
Resolved